addons/sale_timesheet/wizard/sale_make_invoice_advance.py

```diff
@@ -28,6 +29,12 @@ def _compute_invoicing_timesheet_enabled(self):
                 )
             )
 
+    @api.depends('sale_order_ids.tasks_ids.timesheet_ids')
+    def _compute_has_timer_running(self):
+        for inv in self:
+            timesheets = inv.sale_order_ids._origin.tasks_ids.timesheet_ids.filtered(lambda l: not l.validated)
+            inv.has_timer_running = self.env['timer.timer'].sudo().search_count([('res_id', 'in', timesheets.ids), ('res_model', '=', timesheets._name)], limit=1) > 0
```

Comment:
```suggestion
        param_invoiced_timesheet = self.env['ir.config_parameter'].sudo().get_param('sale.invoiced_timesheet', DEFAULT_INVOICED_TIMESHEET)
        if param_invoiced_timesheet == 'approved':
            # Then no need to check if there is some timer running for the 
            # timesheets linked to the SOs because we will just take the 
            # validated ones and so no timer should be running for those ones.
            wizard.has_timer_running = False
            return
        Timesheet = self.env['account.analytic.line']
        for wizard in self:
            timesheet_ids = Timesheet.sudo()._search([
                ('project_id', '!=', False), 
                ('order_id', '=', wizard.sale_order_ids.ids),
                ('validated', '=', False),
            ])
            wizard.has_timer_running = self.env['timer.timer'].sudo().search_count([('res_id', 'in', timesheets.ids), ('res_model', '=', timesheets._name)], limit=1) > 0
```

1. For the following part in my suggestion
```py
        param_invoiced_timesheet = self.env['ir.config_parameter'].sudo().get_param('sale.invoiced_timesheet', DEFAULT_INVOICED_TIMESHEET)
        if param_invoiced_timesheet == 'approved':
            # Then no need to check if there is some timer running for the 
            # timesheets linked to the SOs because we will just take the 
            # validated ones and so no timer should be running for those ones.
            wizard.has_timer_running = False
            return
```
I think the bug is only there is that setting in the configuration of timesheets app is the following one:
![image](https://github.com/odoo/odoo/assets/24781386/bd2a107e-1aa0-499d-ba36-5e5ac1081894)
Since if [we will just take the validated timesheets to link them to the new invoices](https://github.com/odoo/enterprise/blob/master/sale_timesheet_enterprise/models/account.py#L13-L18) then we can assume no timer is running for those timesheets. 😁

2. I search the timesheets in sudo because the salesman could have no access to timesheets app. (No need to get them via `task_ids.timesheet_ids` since the `order_id` field in timesheets model is normally correctly populated to get them) 🙂

3. Also, your code should be in `sale_timesheet_enterprise` module instead of `sale_timesheet` one, because the `timer` feature is an enterprise feature and `validated` field in timesheet model is defined in `timesheet_grid` module (enterprise module). 😬 
Can you move all your changes in `sale_timesheet_enterprise` module?
(Sorry I did not see it during my first review. 😢)